src/seedu/addressbook/Main.java

```diff
@@ -106,11 +106,14 @@ private void recordResult(CommandResult result) {
      * @return result of the command
      */
     private CommandResult executeCommand(Command command)  {
+        command.setData(addressBook, lastShownList);
+        CommandResult result = command.execute();
         try {
-            command.setData(addressBook, lastShownList);
-            CommandResult result = command.execute();
             storage.save(addressBook);
             return result;
+        } catch (StorageOperationException e) {
+            ui.showToUser(e.getMessage());
```

Comment:
Perhaps you may want to be more specific with what you want to show to the user?

Reply:
You don't need to make a new PR, but if you want to show me in class that's fine as well. Alternatively you could push the changes to your w4.5c branch directly?